public/components/trace_analytics/requests/queries/traces_queries.ts

```diff
@@ -557,20 +562,28 @@ export const getCustomIndicesTracesQuery = (
     query: {
       bool: {
         must: [],
-        filter: [
-          {
-            term: {
-              parentSpanId: '', // Data prepper root span doesn't have any parent.
-            },
-          },
-        ],
+        filter: [],
         should: [],
         must_not: [],
       },
     },
     ...(sort && { sort: [{ [sort.field]: { order: sort.direction } }] }),
     track_total_hits: false,
   };
+
+  if (queryMode === 'trace_root_spans') {
+    dataPrepperQuery.query.bool.filter.push({
```

Comment:
Not Related: we should have used an opensource name.

Reply:
updated to `root_spans`